### ISSUE TITLE
iOS - Bugsnag reported a bugsnag crash

### ISSUE THREAD

**Issue by cristianoccazinsp** [bugsnag/bugsnag-js#1803]
### Describe the bug
Just got the following crash in bugsnag, from bugsnag (the irony!). 

```
CrashReporter Key:  a059ba380cb91e11a23719d99fd5054bf55507b9
Hardware Model:     iPad12,2
Process:            zinspector3
Identifier:         com.zinspector.zinspector3
Version:            3.2.35
Role:               Foreground
OS Version:         iOS 15.6


std::bad_alloc: std::bad_alloc

0  zinspector3             bsg_kscrw_i_getBacktrace (BSG_KSCrashReport.c:459:13)
1  zinspector3             bsg_kscrw_i_writeThread (BSG_KSCrashReport.c:909:9)
2  zinspector3             bsg_kscrw_i_writeAllThreads (BSG_KSCrashReport.c:977:17)
3  zinspector3             bsg_kscrw_i_writeTraceInfo (BSG_KSCrashReport.c:1522:9)
4  zinspector3             bsg_kscrashreport_writeStandardReport (BSG_KSCrashReport.c:1472:9)
5  zinspector3             CPPExceptionTerminate() (BSG_KSCrashSentry_CPPException.mm:213:9)
6  libc++abi.dylib         std::__terminate(void (*)())
7  libc++abi.dylib         std::terminate()
8  libdispatch.dylib       __dispatch_client_callout
9  libdispatch.dylib       __dispatch_lane_serial_drain
10 libdispatch.dylib       __dispatch_lane_invoke
11 libdispatch.dylib       __dispatch_workloop_worker_thread
12 libsystem_pthread.dylib __pthread_wqthread
13 libsystem_pthread.dylib _start_wqthread
```

### Steps to reproduce
1. No reproduction steps. App was in foreground and under regular use.

### Environment
* Bugsnag version: "@bugsnag/react-native": "7.17.3"
* Browser framework version (if any): -
* Server framework version (if any): -
* Browser version (e.g. chrome, safari): -
* Device (e.g. iphonex): iOS 15.6 - iPad 9 (Cellular) (iPad12,2)

**Comment by luke-belton** [comment]
Hi @cristianoccazinsp - thanks for raising this with us. We're investigating and will keep you updated on this thread 👍

**Comment by luke-belton** [comment]
Hi @cristianoccazinsp - our investigation showed that rather than this being a crash in Bugsnag, we were actually incorrectly reporting the C++ exception stacktrace, due to Bugsnag being linked dynamically in your app. We've added a [fix](https://github.com/bugsnag/bugsnag-cocoa/pull/1463) for that behaviour, which has just been released in [`v6.22.2`](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/6.22.2) of `bugsnag-cocoa`.

Note that when you upgrade, if the same crash happens again in your app you will see a new error in your dashboard (i.e. it won't group with the existing `std::bad_alloc` error you shared the stacktrace for above). The crash will have a different, but this time correct, stacktrace.

Thanks again for bringing this to our attention!
